pkg/opts/cgroup.go

```diff
@@ -0,0 +1,18 @@
+package opts
+
+import (
+	"context"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/linux/runcopts"
+)
+
+// WithContainerdShimCgroup returns function that sets the containerd
+// shim cgroup path
+func WithContainerdShimCgroup(path string) containerd.NewTaskOpts {
+	return func(_ context.Context, _ *containerd.Client, r *containerd.TaskInfo) error {
```

Comment:
Check options. If it's nil, create a new one; if not, just assign `ShimCgroup` directly.